eclair-core/src/main/scala/fr/acinq/eclair/payment/LocalPaymentHandler.scala

```diff
@@ -85,12 +85,21 @@ class LocalPaymentHandler(nodeParams: NodeParams) extends Actor with ActorLoggin
               log.warning(s"received payment with amount too large for paymentHash=${htlc.paymentHash} amountMsat=${htlc.amountMsat}")
               sender ! CMD_FAIL_HTLC(htlc.id, Right(IncorrectPaymentAmount), commit = true)
             case _ =>
-              log.info(s"received payment for paymentHash=${htlc.paymentHash} amountMsat=${htlc.amountMsat}")
-              // amount is correct or was not specified in the payment request
-              nodeParams.paymentsDb.addPayment(Payment(htlc.paymentHash, htlc.amountMsat, Platform.currentTime / 1000))
-              sender ! CMD_FULFILL_HTLC(htlc.id, paymentPreimage, commit = true)
-              context.system.eventStream.publish(PaymentReceived(MilliSatoshi(htlc.amountMsat), htlc.paymentHash, htlc.channelId))
-              context.become(run(hash2preimage - htlc.paymentHash))
+              paymentRequest.expiry match {
+                case Some(expiry: Long) if (expiry+paymentRequest.timestamp <  Platform.currentTime / 1000) =>
```

Comment:
nitpick: specifying the type of expiry is not needed here, and we'd use space around the `+`